### DISCUSSION TITLE
Autocomplete and javascript maps api using OpenStreetMap Data

### DISCUSSION THREAD

**Discussion by andrea989898** [osm-search/Nominatim#3037] [Q&A]
Hi everyone,

I would like to ask for your help in organizing the general architecture of the following project: I want to build a service that allows me to call an autocomplete API for addresses and one that allows me to display a javascript map using OpenStreetMaps data (I only need it for Italy and Switzerland). Everything will be hosted on Azure.

I searched the internet and found these projects:
- Nominatim, which allows me to create the autocomplete API.
- protomaps seems like a good project for lower costs, I can store the tiles in Azure blob storage.

Is there any way to share data between the two? What should I use/do?

I will use the autocomplete for address search, while the maps to show stores on the map with custom markers.

If you have any other ideas or want to start a discussion on this, it would be fantastic.
Thanks and have a good

**Comment by otbutz** [comment]
For autocomplete address search: https://photon.komoot.io/

**Comment by AshadeSamson** [comment]
From my understanding, you want to display some results on a map based on the data a user searches, then you want to use autocomplete to help them complete their search entry, Right?

You can use the leaflet.js library for the maps. check here https://leafletjs.com/
